torch/csrc/distributed/autograd/context/context.h

```diff
@@ -130,6 +141,20 @@ class TORCH_API DistAutogradContext {
 };
 
 using ContextPtr = std::shared_ptr<DistAutogradContext>;
+using ContextWeakPtr = std::weak_ptr<DistAutogradContext>;
+
+// This class stores a weak_ptr to the current dist_autograd context in a thread
+// local variable so that a reader can retrieve the context.
+class TORCH_API ThreadLocalDistAutogradContext {
```

Comment:
nit: Since we already have another thread local dist autograd context, lets make it clear this is meant only for post hooks. Let's call it something like `BackwardHookThreadLocalContext`.  If you can find a shorter name that conveys the same even better :)

Reply:
It's actually a generic thread local. I don't see from its definition that it's only for post hooks, especially it's defined together with `DistAutogradContext`.

Reply:
We should at least mention how this thread local is different from the one here: https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/autograd/context/container.cpp#L17. It could be confusing for developers to see that we have two thread locals which sound similar.

Reply:
Added comments